parquet-column/src/test/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReaderTest.java

```diff
@@ -0,0 +1,223 @@
+      byte[] byteData = {(byte) 0x00, (byte) 0x00, (byte) 0x10, (byte) 0x40};
+      ByteBuffer buffer = ByteBuffer.wrap(byteData);
+      ByteBufferInputStream stream = ByteBufferInputStream.wrap(buffer);
+
+      ByteStreamSplitValuesReaderForFloat reader = new ByteStreamSplitValuesReaderForFloat();
+      reader.initFromPage(1, stream);
+      float f = reader.readFloat();
+      assertEquals(2.25f, f, 0.0f);
+    }
+
+    @Test
+    public void TestSmallBuffer() throws Exception {
+      byte[] byteData = {(byte) 0x40, (byte) 0x00, (byte) 0x80, (byte) 0x40, (byte) 0x05, (byte) 0x84,
+        (byte) 0xc5, (byte) 0xbd, (byte) 0x32, (byte) 0xc2, (byte) 0x41, (byte) 0x42};
+      ByteBuffer buffer = ByteBuffer.wrap(byteData);
+      ByteBufferInputStream stream = ByteBufferInputStream.wrap(buffer);
+
+      ByteStreamSplitValuesReaderForFloat reader = new ByteStreamSplitValuesReaderForFloat();
+      reader.initFromPage(3, stream);
+      assertEquals(-98.62548828125, reader.readFloat(), 0.0f);
+      assertEquals(23.62744140625, reader.readFloat(), 0.0f);
+      assertEquals(44.62939453125, reader.readFloat(), 0.0f);
+    }
+
+    @Test
+    public void TestRandomInput() throws Exception {
+      Random rand = new Random(1337);
+      final int numElements = 256;
+      byte[] byteData = new byte[numElements * 4];
+      float[] values = new float[numElements];
+      for (int i = 0; i < numElements; ++i) {
+        float f = rand.nextFloat() * 1024.0f;
+        values[i] = f;
+        int fAsInt = Float.floatToIntBits(f);
+        byteData[i] = (byte) (fAsInt & 0xFF);
+        byteData[numElements + i] = (byte) ((fAsInt >> 8) & 0xFF);
+        byteData[numElements * 2 + i] = (byte) ((fAsInt >> 16) & 0xFF);
+        byteData[numElements * 3 + i] = (byte) ((fAsInt >> 24) & 0xFF);
+      }
+      ByteBuffer buffer = ByteBuffer.wrap(byteData);
+      ByteBufferInputStream stream = ByteBufferInputStream.wrap(buffer);
+
+      ByteStreamSplitValuesReaderForFloat reader = new ByteStreamSplitValuesReaderForFloat();
+      reader.initFromPage(numElements, stream);
+
+      for (int i = 0; i < numElements; ++i) {
+        float f = reader.readFloat();
+        assertEquals(values[i], f, 0.0f);
+      }
+    }
+
+    @Test
+    public void TestExtraReads() throws Exception {
+      byte[] byteData = {(byte) 0x00, (byte) 0x00, (byte) 0x10, (byte) 0x40};
+      ByteBuffer buffer = ByteBuffer.wrap(byteData);
+      ByteBufferInputStream stream = ByteBufferInputStream.wrap(buffer);
+
+      ByteStreamSplitValuesReaderForFloat reader = new ByteStreamSplitValuesReaderForFloat();
+      reader.initFromPage(1, stream);
+      float f = reader.readFloat();
+      assertEquals(2.25f, f, 0.0f);
+      boolean exceptionWasThrown;
+      try {
+        reader.readFloat();
+        exceptionWasThrown = false;
+      } catch (ParquetDecodingException ex) {
+        exceptionWasThrown = true;
+      }
+      assertEquals(true, exceptionWasThrown);
```

Comment:
Suggestion:
If you are expecting an exception, you might specify it as an argument of the Test annotation. This way you do not have to try-catch and do any additional assertions.
If you would like to verify that the exception is thrown from the exact code line you might use try-catch and simply `Assert.fail(msg)` in the try block after the line you are testing.

Reply:
Done